libs/vm-modules/tests/unit/ml_charge_estimation.cpp

```diff
@@ -0,0 +1,173 @@
+
+/* This fixture is used for consistency tests of the charge amount estimation
+ * for VMModel objects                                                        */
+
+class MLChargeEstimationTests : public ::testing::Test
+{
+public:
+  VMPtr        vm;
+  Ptr<VMModel> model;
+
+  void SetUp() override
+  {
+    // setup the VM
+    auto module = VMFactory::GetModule(VMFactory::USE_SMART_CONTRACTS);
+    vm          = std::make_unique<VM>(module.get());
+
+    // create a VMModel
+    model = CreateSequentialModel();
+  }
+
+  Ptr<String> CreateString(std::string const &str)
+  {
+    return Ptr<String>{new String{vm.get(), str}};
+  }
+
+  Ptr<Array<uint64_t>> CreateArray(std::vector<uint64_t> const &values)
+  {
+    std::size_t          size = values.size();
+    Ptr<Array<uint64_t>> array =
+        vm->CreateNewObject<Array<uint64_t>>(vm->GetTypeId<uint64_t>(), static_cast<int32_t>(size));
+    ;
+
+    for (std::size_t i{0}; i < size; ++i)
+    {
+      array->elements[i] = values[i];
+    }
+
+    return array;
+  }
+
+  // Ptr<VMTensor> CreateTensor(Ptr<Array<uint64_t>> const& shape)
+  Ptr<VMTensor> CreateTensor(std::vector<uint64_t> const &shape)
+  {
+    return vm->CreateNewObject<VMTensor>(shape);
+  }
+
+  Ptr<VMModel> CreateSequentialModel()
+  {
+    auto model_category = CreateString("sequential");
+    return vm->CreateNewObject<VMModel>(model_category);
+  }
+};
+
+//////////////////////////////////////////////////
+// I. Charge amounts relative consistency tests //
+//////////////////////////////////////////////////
+
+// I.1 LayerAddDense
+
+// 1.1. Layer size impact on AddLayerDense() Charge amount
+// 1.1. Layer size impact on AddLayerDenseActivation() Charge amount
+
+TEST_F(MLChargeComputationTests, )
+{
+  using SizeRef      = fetch::math::SizeType const &;
+  using StringPtrRef = fetch::vm::Ptr<fetch::vm::String> const &;
+
+  // set up data and labels
+  std::vector<uint64_t> data_shape{10, 1000};
+  std::vector<uint64_t> label_shape{1, 1000};
+  auto                  data  = CreateTensor(data_shape);
+  auto                  label = CreateTensor(label_shape);
+
+  // set up a model
+  auto model = CreateSequentialModel();
+  model->AddLayer<SizeRef, SizeRef, StringPtrRef>(
+      CreateString("dense"), 10, 10,
+      CreateString("relu"));  // input_size, hidden_1_size
+  model->AddLayer<SizeRef, SizeRef, StringPtrRef>(
+      CreateString("dense"), 10, 10,
+      CreateString("relu"));                                        // hidden_1_size, hidden_2_size
+  model->AddLayer<SizeRef, SizeRef>(CreateString("dense"), 10, 1);  // hidden_2_size, label_size
+  model->CompileSequential(CreateString("mse"), CreateString("adam"));
+
+  // train the model
+  model->Fit(data, label, 32);  // batch_size
+
+  // get loss value
+  /*auto loss* = */ model->Evaluate();
+
+  // make a prediction
+  /*Ptr<VMTensor> prediction =*/model->Predict(data);
+
+  ASSERT_TRUE(1 != 1);
+}
+
+// I.2 CompileSequential
```

Comment:
probably better if we move all these comments into Jira tasks with the relevant descriptions and remove them from the codebase - since they're not going to be implemented in this PR

Reply:
yes